radix-engine/src/blueprints/locker/blueprint.rs

```diff
@@ -0,0 +1,764 @@
+        let badge_address = ResourceAddress::new_or_panic(badge_address.as_node_id().0);
+
+        let (_, badge) = api
+            .call_function(
+                RESOURCE_PACKAGE,
+                FUNGIBLE_RESOURCE_MANAGER_BLUEPRINT,
+                FUNGIBLE_RESOURCE_MANAGER_CREATE_WITH_INITIAL_SUPPLY_IDENT,
+                scrypto_encode(&FungibleResourceManagerCreateWithInitialSupplyInput {
+                    owner_role: OwnerRole::Updatable(rule!(require(badge_address))),
+                    track_total_supply: true,
+                    divisibility: 0,
+                    resource_roles: Default::default(),
+                    metadata: metadata! {
+                        init {
+                            "name" => "Account Locker Admin Badge".to_owned(), locked;
+                        }
+                    },
+                    address_reservation: Some(badge_reservation),
+                    initial_supply: dec!(1),
+                })
+                .unwrap(),
+            )
+            .map(|rtn| {
+                scrypto_decode::<FungibleResourceManagerCreateWithInitialSupplyOutput>(&rtn)
+                    .unwrap()
+            })?;
+
+        // Preparing all of the roles and rules.
+        let rule = rule!(require(badge_address));
+        let recoverer_rule = match allow_forceful_withdraws {
+            true => rule.clone(),
+            false => rule!(deny_all),
+        };
+
+        Self::instantiate(
+            AccountLockerInstantiateInput {
+                owner_role: OwnerRole::Updatable(rule.clone()),
+                storer_role: rule.clone(),
+                storer_updater_role: rule.clone(),
+                recoverer_role: recoverer_rule.clone(),
+                recoverer_updater_role: recoverer_rule,
+                address_reservation: None,
+            },
+            api,
+        )
+        .map(|rtn| (rtn, badge))
+    }
+
+    fn store<Y>(
+        AccountLockerStoreInput { claimant, bucket }: AccountLockerStoreInput,
+        api: &mut Y,
+    ) -> Result<AccountLockerStoreOutput, RuntimeError>
+    where
+        Y: ClientApi<RuntimeError>,
+    {
+        // Bucket info.
+        let (resource_address, resource_specifier) = bucket_to_resource_specifier(&bucket, api)?;
+
+        // Store in the vault.
+        Self::with_vault_create_on_traversal(
+            claimant.0,
+            resource_address,
+            api,
+            |mut vault, api| vault.put(bucket, api),
+        )??;
+
+        // Emit an event with the stored resource
+        Runtime::emit_event(
+            api,
+            StoreEvent {
+                claimant,
+                resource_address,
+                resources: resource_specifier,
+            },
+        )?;
+
+        Ok(())
+    }
+
+    fn store_batch<Y>(
+        AccountLockerStoreBatchInput { claimants, bucket }: AccountLockerStoreBatchInput,
+        api: &mut Y,
+    ) -> Result<AccountLockerStoreBatchOutput, RuntimeError>
```

Comment:
Yup thought about this one:

* `Vec<(Bucket, GlobalAddress)>` - Batching with this will mean that an airdropping manifest will have tons of take from worktop instructions making it slightly less readable and more difficult to construct. 
* `(Bucket, Vec<Claimant>)` - Feels conceptually like the thing that users would want to do. Here is a bucket and here is how it is split across the recipients.